cvmfs/dns.cc

```diff
@@ -1153,9 +1153,9 @@ void HostfileResolver::ParseHostFile() {
     if (hash_pos != string::npos) line = line.substr(0, hash_pos);
 
     // First token is an IP address
-    int ip_start_pos, ip_end_pos;
-    sscanf(line.c_str(), " %n%*s%n", &ip_start_pos, &ip_end_pos);
-    if (ip_start_pos == ip_end_pos)
+    int ip_start_pos, ip_end_pos, scan_result;
+    scan_result = sscanf(line.c_str(), " %n%*s%n", &ip_start_pos, &ip_end_pos);
+    if (scan_result == EOF || ip_start_pos == ip_end_pos)
```

Comment:
I think this should test for `scan_result == 2` (instead of only EOF)

Reply:
I tested sscanf on my computer and it returns 0 even when all of the variables are filled. It seems that both `%n` and `%*s` do not count as filled variables.